runtime/jvmti/jvmtiThread.c

```diff
@@ -922,7 +938,24 @@ jvmtiGetThreadLocalStorage(jvmtiEnv* env,
 
 		rc = getVMThread(currentThread, thread, &targetThread, TRUE, TRUE);
 		if (rc == JVMTI_ERROR_NONE) {
-			rv_data = THREAD_DATA_FOR_VMTHREAD((J9JVMTIEnv *) env, targetThread)->tls;
+			J9JVMTIEnv *j9env = (J9JVMTIEnv *)env;
```

Comment:
I believe this block is needed only when `NULL != data_ptr`

Reply:
The `ENSURE_NON_NULL(data_ptr);` line above already checks this